# Genomics
6110 assignment
# Introduction
Genome assembly is the process of piecing together fragmented sequencing data into a single biological sequence which is a prerequisite for meaningful genomic analysis. Despite the accessibility of modern sequencing platforms, the production of accurate, gap-free genomes remains a complex task. The quality of these assemblies directly impacts the reliability of downstream investigations, particularly in comparative genomics and the detection of genetic markers associated with pathogenicity and drug resistance (Whibley et al., 2021). 
The objective of this assignment is to assemble the genome of Salmonella enterica using long-read sequencing data generated by an Oxford Nanopore Technologies (ONT) platform and to compare the resulting consensus genome to a reference genome obtained from the National Center for Biotechnology Information (NCBI). 
Salmonella enterica is a clinically and epidemiologically important bacterial species responsible for a wide range of foodborne illnesses worldwide. Precise genomic reconstruction allows for the detection of minute strain-level variations, which is fundamental for accurate source attribution and the phylogenetic tracking of outbreaks (Pightling et al., 2018). 
The primary advantage of long-read sequencing is its ability to resolve complex genomic repeats by anchoring them to unique flanking sequences. In bacterial genomics, achieving a complete, single-contig assembly usually requires reads that exceed 8kb. This length is critical because it allows the sequencer to bridge the ribosomal RNA (rRNA) operons, which are generally the largest identical repetitive regions found in bacteria (Cock et al., 2009). To address the limitations of short-read platforms, the field has shifted toward 'third-generation' technologies specifically engineered to generate substantially longer sequencing reads (Marx, 2023).
This requirement is currently best met by Oxford Nanopore's R10 flow cells, which combine the continuity of ultra-long reads with high-fidelity base-calling (Q20+). This combination effectively removes the need for short-read error correction, streamlining the assembly pipeline for bacterial genomes like Salmonella (Senol Cali et al., 2019). 
Despite these advantages, Nanopore sequencing also presents important challenges including sequencing errors particularly in homopolymer regions as well as balancing contiguity and correctness (Wick et al., 2019). These errors can lead to insertions, deletions, or mismatches in the assembled genome if not properly addressed. Consequently, genome assembly using Nanopore data requires specialized algorithms that are robust to noisy reads, as well as post-assembly polishing steps to improve base-level accuracy. These limitations highlight the importance of combining de novo assembly with reference-based analysis to obtain a more complete and accurate representation of the genome.
## Method Comparison 
Generating accurate de novo assemblies from long-read platforms requires appropriate software. The choice of analysis tools is the primary determinant of the final output's quality, influencing everything from the quality, contiguity, and biological accuracy of the resulting genomes. For long-read bacterial genome assembly using Oxford Nanopore Technologies (ONT) data, assemblers such as Flye, Canu, Shasta, and Raven are commonly used (Kolmogorov et al., 2019). In a study conducted by Cosma et al. (2023), Flye was the assembler that scored consistently good in most evaluation categories for assemblies of reads in ONT data. According to a study conducted in 2025 by (Kumar et al., 2025), it was concluded that when maximizing the length of contiguous fragments, tools such as NextDenovo and NECAT are often the top performers. In contrast, Flye is frequently recommended for its ability to strike a balance between computational speed and base-level accuracy. While Canu remains a powerful option for deep, correction-intensive assembly. Unicycler is distinctively noted for its ability to resolve circular chromosomes.
# Methodology
## Data preprocessing and Quality Assessment
Raw long-read sequencing data for Salmonella enterica was downloaded from NCBI database with accession number SRR32410565.1 The data was extracted using SRA Toolkit (v2.9.6) which converted SRA format into FASTQ sequences. Initial quality assessment was conducted using NanoPlot (v1.4.2) to characterize read length distribution and quality score metrics (De Coster et al., 2023). In order to ensure quality, the raw reads were subjected to filtering using Filtlong (v0.3.1). So reads shorter than 1,000 base pairs (bp) or with a mean quality score below Q10 were discarded to remove truncated fragments and low-accuracy sequences that could introduce graph ambiguities. Afterwards NanoPlot was rerun again on the filtered reads to confirm the differences and validate the removal of noise.
```bash
fasterq-dump SRR32410565.1 --threads 4
NanoPlot --fastq SRR32410565.1.fastq --outdir qc_check
Filtering
filtlong --min_length 1000 --min_mean_q 10 SRR32410565.1.fastq > filtered_reads.fastq
Nanoplot of filtered reads
NanoPlot --fastq filtered_reads.fastq --outdir qc_filtered
```
## Genome Assembly
The filtered reads were assembled using Flye (v2.9.6). The pipeline was executed in --nano-hq mode to leverage the improved accuracy profile of R10 pore chemistry. An expected genome size of 4.8 Mb was specified to guide the coverage estimation and graph resolution parameters (Wick & Holt, 2022).
```bash
flye --nano-hq filtered_reads.fastq --out-dir flye_assembly --genome-size 4.8m --threads 4
```
## Assembly Polishing
Following assembly, the draft genome was polished using Medaka a neural-network–based polishing tool designed specifically for Oxford Nanopore data with the model to correct homopolymer errors and improve base-level accuracy (Sereika et al., 2022). 
## Genome Alignment and Variant Calling
The reference genome was downloaded from NCBI and thereafter aligned with the polished assembly using Minimap2 (v 2.24) (Jain et al., 2020). The resulting alignment was sorted and indexed using Samtools (v1.16) for downstream analysis. Structural variants (SVs), including insertions, deletions, and inversions, were detected using SVIM-asm (v1.0.3) in haploid mode.
```bash
minimap2 -ax asm5 reference.fasta medaka_out/consensus.fasta > alignment_polished.sam
samtools view -bS alignment_polished.sam | samtools sort -o alignment_polished_sorted.bam samtools index alignment_polished_sorted.bam
svim-asm haploid/home/sodiq-dada alignment_polished_sorted.bam reference.fasta
```
## Visualization
Genome alignments and identified variants were visualized using IGV (Integrative Genomics Viewer) (v2.18) (Robinson et al., 2023)

# Results and Discussion
## Data Quality and Filtering 
The raw read “SRR32410565” before subjecting it to filtering had 391,551 reads. Inspection via NanoPlot revealed a variable length distribution typical of raw long-read output. While Nanopore sequencing offers the distinct advantage of spanning long repetitive regions (Jain et al., 2018), it is historically prone to higher error rates compared to short-read platforms. In order to remove unnecessary noise, filtlong was applied to remove reads shorter than 1000bp and with a mean quality score below Q10.
After filtering, the dataset was reduced to 387,807 high-quality reads. While this represented a minor reduction in total read count (<1%), it significantly improved the mean read quality, removing the "noise" floor of fragmented sequences as seen in Figure 1.
![image alt](https://github.com/Sodjoh/Genomics/blob/9eee75778a6729c3b085ee0edb21285446073c52/LengthvsQualityScatterPlot_dot.png)
                
Figure 1: The scatter plot displays read length (x-axis) versus mean quality score (y-axis) before and after filtering. The absence of data points in the lower-left quadrant (<1kb, <Q10) confirms the successful removal of low-quality fragments.

## De Novo Genome Assembly
The filtered reads were assembled with flye and polished with medaka. The resulting assembly metrics gotten from QUAST indicate a highly contiguous reconstruction of the Salmonella enterica genome. The high N50 value and low contig count confirm that the bacterial chromosome was assembled into a nearly complete, continuous sequence. Visualization of the assembly graph using Bandage revealed that the Salmonella genome was assembled into two distinct components (Figure 2). The primary chromosomal component is characterized by two unique contig loops connected by a central, high-coverage repeat node. This structure indicates a near-complete assembly where a long repetitive element (likely an rRNA operon) remains unresolved. Additionally, a small, circular, extrachromosomal component was identified, representing a plasmid. This structure is a common artifact in bacterial assembly, often caused by rRNA operons (16S/23S) that exceed the read length of the sequencing platform (Wick et al., 2017) Despite this, the connectivity confirms the recovery of the complete chromosomal sequence. Additionally, the identification of a closed, extrachromosomal circular contig suggests the presence of a plasmid, a common feature in Salmonella serovars known to harbor virulence determinants (McClelland et al., 2001)


# References
Cock, P. J. A., Antao, T., Chang, J. T., Chapman, B. A., Cox, C. J., Dalke, A., Friedberg, I., Hamelryck, T., Kauff, F. and Wilczynski, B. (2009). Biopython: freely available Python tools for computational molecular biology and bioinformatics. Bioinformatics, 25(11), 1422.

Cosma, B.-M., Shirali Hossein Zade, R., Jordan, E. N., van Lent, P., Peng, C., Pillay, S. and Abeel, T. (2023). Evaluating long-read de novo assembly tools for eukaryotic genomes: insights and considerations. GigaScience, 12, giad100.

Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M. O., Whitwham, A., Keane, T., McCarthy, S. A. and Davies, R. M. (2021). Twelve years of SAMtools and BCFtools. GigaScience 10, giab008. In.

De Coster, W. and Rademakers, R. (2023). NanoPack2: population-scale evaluation of long-read sequencing data. Bioinformatics, 39(5), btad311.

Jain, C., Rhie, A., Zhang, H., Chu, C., Walenz, B. P., Koren, S. and Phillippy, A. M. (2020). Weighted minimizer sampling improves long read mapping. Bioinformatics, 36(Supplement_1), i111-i118.

Kumar, M. S., Krishna, M. B., Soman, K. P., Stanley, J., Pourmand, N., Suravajhala, P. and Babu, T. G. S. (2025). Benchmarking long-read assembly tools and preprocessing strategies for bacterial genomes: A case study on E. coli DH5α. Biotechnology Reports, e00931.

Marx, V. (2023). Method of the year: long-read sequencing. Nature methods, 20(1), 6-11.

Pightling, A. W., Pettengill, J. B., Luo, Y., Baugher, J. D., Rand, H. and Strain, E. (2018). Interpreting whole-genome sequence analyses of foodborne bacteria for regulatory applications and outbreak investigations. Frontiers in microbiology, 9, 1482.

Robinson, J. T., Thorvaldsdottir, H., Turner, D. and Mesirov, J. P. (2023). igv.js: an embeddable JavaScript implementation of the Integrative Genomics Viewer (IGV). Bioinformatics, 39(1), btac830.

Senol Cali, D., Kim, J. S., Ghose, S., Alkan, C. and Mutlu, O. (2019). Nanopore sequencing technology and tools for genome assembly: computational analysis of the current state, bottlenecks and future directions. Briefings in bioinformatics, 20(4), 1542-1559.

Sereika, M., Kirkegaard, R. H., Karst, S. M., Michaelsen, T. Y., Sørensen, E. A., Wollenberg, R. D. and Albertsen, M. (2022). Oxford Nanopore R10. 4 long-read sequencing enables the generation of near-finished bacterial genomes from pure cultures and metagenomes without short-read or reference polishing. Nature methods, 19(7), 823-826.

Whibley, A., Kelley, J. L. and Narum, S. R. (2021). The changing face of genome assemblies: Guidance on achieving high‐quality reference genomes. In: Wiley Online Library.

Wick, R. R. and Holt, K. E. (2021). Benchmarking of long-read assemblers for prokaryote whole genome sequencing. F1000Research, 8, 2138.
