# Genomics
6110 assignment
# Introduction
Genome assembly is the process of piecing together fragmented sequencing data into a single biological sequence which is a prerequisite for meaningful genomic analysis. Despite the accessibility of modern sequencing platforms, the production of accurate, gap-free genomes remains a complex task. The quality of these assemblies directly impacts the reliability of downstream investigations, particularly in comparative genomics and the detection of genetic markers associated with pathogenicity and drug resistance (Whibley et al., 2021). 
The objective of this assignment is to assemble the genome of Salmonella enterica using long-read sequencing data generated by an Oxford Nanopore Technologies (ONT) platform and to compare the resulting consensus genome to a reference genome obtained from the National Center for Biotechnology Information (NCBI). 
Salmonella enterica is a clinically and epidemiologically important bacterial species responsible for a wide range of foodborne illnesses worldwide. Precise genomic reconstruction allows for the detection of minute strain-level variations, which is fundamental for accurate source attribution and the phylogenetic tracking of outbreaks (Pightling et al., 2018). 
The primary advantage of long-read sequencing is its ability to resolve complex genomic repeats by anchoring them to unique flanking sequences. In bacterial genomics, achieving a complete, single-contig assembly usually requires reads that exceed 8kb. This length is critical because it allows the sequencer to bridge the ribosomal RNA (rRNA) operons, which are generally the largest identical repetitive regions found in bacteria (Cock et al., 2009). To address the limitations of short-read platforms, the field has shifted toward 'third-generation' technologies specifically engineered to generate substantially longer sequencing reads (Marx, 2023).
This requirement is currently best met by Oxford Nanopore's R10 flow cells, which combine the continuity of ultra-long reads with high-fidelity base-calling (Q20+). This combination effectively removes the need for short-read error correction, streamlining the assembly pipeline for bacterial genomes like Salmonella (Senol Cali et al., 2019). 
Despite these advantages, Nanopore sequencing also presents important challenges including sequencing errors particularly in homopolymer regions as well as balancing contiguity and correctness (Wick et al., 2019). These errors can lead to insertions, deletions, or mismatches in the assembled genome if not properly addressed. Consequently, genome assembly using Nanopore data requires specialized algorithms that are robust to noisy reads, as well as post-assembly polishing steps to improve base-level accuracy. These limitations highlight the importance of combining de novo assembly with reference-based analysis to obtain a more complete and accurate representation of the genome.
